lru.go

```diff
@@ -36,12 +75,30 @@ func (lc *LRUCache) Peek(key string) (interface{}, error) {
 	if !found {
 		return nil, ErrNotFound
 	}
-	return value, nil
+	if lc.compressionEngine == nil {
+		return value, nil
+	}
+
+	return lc.decompress(key, value)
 }
 
 // Set stores given key-value pair into cache
 func (lc *LRUCache) Set(key string, value interface{}) error {
-	lc.lru.Add(key, value)
+	if lc.compressionEngine == nil {
```

Comment:
👍 Great, I was worried that with adding of compression and marshalling we will lose the possibility of 0-copy cache, but this should work.